src/core/KuzzleEventEmitter.ts

```diff
@@ -54,7 +84,50 @@ export class KuzzleEventEmitter {
     return this;
   }
 
-  on(eventName, listener) {
+  on(
+    eventName:
+      | "connected"
+      | "reconnected"
+      | "tokenExpired"
+      | "reAuthenticated"
+      | "websocketRenewalStart"
+      | "websocketRenewalDone",
+    listener: () => void
+  ): this;
+  on(
+    eventName: "logoutAttempt",
+    listener: (status: { success: true }) => void
+  ): this;
+  on(
+    eventName: "loginAttempt",
+    listener: (data: { success: boolean; error: string }) => void
+  ): this;
+  on(eventName: "discarded", listener: (request: RequestPayload) => void): this;
+  on(
+    eventName: "disconnect" | "disconnected",
```

Comment:
The `disconnect` event is only emitted by the protocol not the SDK directly, the SDK only emit `disconnected`

Reply:
@Shiranuit Alright thanks. 

In fact this `test/kuzzle-sdk-test.ts`  is not meant to be compiled, but let's do this.
About events, we've got a problem which is I need to type all events because events are triggered by the same event listener and same `on` method, even if they are considered private otherwise the code is inconsistent.

I can only suggest two changes : 
1. Allowing listening on private events and marking them as internal and not stable.
2. Expose a new function `onEvent` which is restricted to "public" events. And mark the original `on` as private.

Maybe there is another way to achieve this, but I'll need your insight.
@Shiranuit @Aschen

Reply:
@Shiranuit It looks like I've been able to play with step 1